tests/periph_wdg/tests/01-run.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2019 Inria
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import sys
+import json
+from testrunner import run
+
+# We test only up to 10ms, with smaller times mcu doesn't have time to
+# print sysyem time before resetting
+reset_times_us = [1e4, 5e4, 1e5, 5e5, 1e6]
+
+
+def get_reset_time(child):
+    reset_time = 0
+    reset = False
+    while not reset:
+        line = child.readline()
+        if "main(): This is RIOT!" in line:
+            break
+        jason = line.strip().split("json:")
+        reset_time = json.loads(jason[1])
+    return reset_time['time']
+
+
+def get_start_time(child):
+    child.readline()
+    child.readline()
+    line = child.readline().strip().split("json:")
+    start_time = json.loads(line[1])
+    return start_time['starttime']
+
+
+def testfunc(child):
+    for rst_time in reset_times_us:
+        child.sendline("setup 0 {}".format(rst_time))
+        child.expect_exact("[wdg]: - ")
+        child.expect_exact(" -")
+
+        if child.before != "invalid configuration time":
+            child.sendline("startloop")
+            start_time = get_start_time(child)
+            reset_time = get_reset_time(child)
```

Comment:
I find `get_reset_time` overly complex. This could be much much simpler using `child.expect` and regex to get the reset time. I also don't like the use of json in the output of the application.
Have a look at how numerical values are retrieved using regexp in [the python script periph_rtt](https://github.com/RIOT-OS/RIOT/blob/master/tests/periph_rtt/tests/01-run.py#L20). This can be done the same way (not tested):
```python
child.expect(r'  Reset time: (\d+)')
reset_time = int(child.match.group(1))
```

Same for start time:
```python
child.expect(r'  Start time: (\d+)')
start_time = int(child.match.group(1))
```

Reply:
@aabadie  Changed it accordingly, what do you think about it now?